### PR TITLE
Added SAI-Challenger setup.

### PR DESCRIPTION
**Pull request by maksymhedeon**
1. Added SAI-Challenger submodule.
2. Added SAI-Challenger basic test.
3. Changed Makefile to build/start/stop SAI-Challenger.

### CODE REVIEW

File: test/test-cases/challenger_basic_test/conftest.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import logging
+import os
+import sys
+import traceback
+import pytest
+from sai_environment import init_setup
+
+curdir = os.path.dirname(os.path.realpath(__file__))
+commondir = os.path.join(curdir, '../common')
```
Comment (plvisiondevs) [inline test/test-cases/challenger_basic_test/conftest.py]: "common" content should be installed in the docker using ```python setup.py install```

File: dash-pipeline/Makefile [hunk 2]
```diff
@@ -496,6 +496,48 @@ deploy-ixiac:
 undeploy-ixiac:
 	cd ../test/third-party/traffic_gen && ./undeploy_ixiac.sh
 
+###############################
+# SAI-CHALLENGER TARGETS
+###############################
+
+SAI_CHALLENGER_PATH = $(PWD)/../test/SAI-Challenger.OCP
+
+sai-challenger-build:sai-challenger-client-build sai-challenger-server-build
```
Comment (plvisiondevs) [inline dash-pipeline/Makefile:506]: Please add a comment that the server build is a temporary staff and has to be removed in the final version.
Add this comment to all server-related commands.

File: test/test-cases/challenger_basic_test/conftest.py (hunk 1 above)
Comment (plvisiondevs) [reply test/test-cases/challenger_basic_test/conftest.py]: If I'm not mistaken, this is by default

**Review by plvisiondevs — CHANGES_REQUESTED** [review]

File: dash-pipeline/Makefile [hunk 3]
```diff
@@ -496,6 +496,55 @@ deploy-ixiac:
...
+	cd $(SAI_CHALLENGER_PATH) && ./build.sh -i client
+
+sai-challenger-client-start:
+#	cd $(SAI_CHALLENGER_PATH) && ./run.sh -i client -r
+	docker run --name sc-client-run \
+        -v $(SAI_CHALLENGER_PATH):/sai-challenger \
+        --cap-add=NET_ADMIN \
+        --device /dev/net/tun:/dev/net/tun \
+        --rm \
+        -v $(SAI_CHALLENGER_PATH)/../test-cases/challenger_basic_test/:/sai-challenger/tests \
```
Comment (elchupanebrej) [inline dash-pipeline/Makefile:527]: Please add volume with dash tests. Please pay attention that dash tests will be stored in Dash repository, so please move  config here from SAI-Challenger repository

File: dash-pipeline/Makefile [hunk 4]
```diff
@@ -496,6 +496,55 @@ deploy-ixiac:
...
+sai-challenger-build:sai-challenger-client-build sai-challenger-server-build
+
+# SAI-C SERVER STAFF IS TEMPORARY AND WILL BE REMOVED LATER
+sai-challenger-start:sai-challenger-client-start sai-challenger-server-start sai-challenger-create-veth
+
+# SAI-C SERVER STAFF IS TEMPORARY AND WILL BE REMOVED LATER
+sai-challenger-stop:sai-challenger-client-stop sai-challenger-server-stop
+
+sai-challenger-test:
+	cd $(SAI_CHALLENGER_PATH) && ./exec.sh -i client pytest --setup=../setups/sai_dpu_client_server.json -v -k "test_dpu_basic"
```
Comment (elchupanebrej) [inline dash-pipeline/Makefile:515]: Please provide rootdir for pytest call

**Pull request merged**